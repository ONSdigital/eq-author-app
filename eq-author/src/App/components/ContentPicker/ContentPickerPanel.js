import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import ScrollPane from "components/ScrollPane";
import { colors, radius } from "constants/theme";

const StyledScrollPane = styled(ScrollPane)`
  padding: 0.25em 0;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

const PanelWrapper = styled.div`
  padding: 0.6em 0.5em;
  position: relative;
  overflow: auto;
  border: 1px solid ${colors.lightGrey};
  border-top: none;
  border-radius: 0 0 ${radius} ${radius};
  display: flex;
  flex-grow: 1;
`;

const ContentPickerPanel = ({ id, open, labelledBy, children }) => {
  if (!open) {
    return null;
  }
  return (
    <PanelWrapper
      id={id}
      role="tabpanel"
      aria-hidden={!open}
      aria-labelledby={labelledBy}
    >
      <StyledScrollPane>{children}</StyledScrollPane>
    </PanelWrapper>
  );
};

ContentPickerPanel.propTypes = {
  id: PropTypes.string.isRequired,
  labelledBy: PropTypes.string.isRequired,
  open: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired
};

export default ContentPickerPanel;
