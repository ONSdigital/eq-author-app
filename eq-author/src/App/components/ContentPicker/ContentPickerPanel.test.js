import React from "react";
import { shallow } from "enzyme";

import ContentPickerPanel from "App/components/ContentPicker/ContentPickerPanel";

describe("ContentPickerPanel", () => {
  const props = {
    id: "1",
    open: true,
    labelledBy: "some-label"
  };
  it("should render", () => {
    const wrapper = shallow(
      <ContentPickerPanel {...props}>Panel contents</ContentPickerPanel>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should not render when not open", () => {
    const wrapper = shallow(
      <ContentPickerPanel {...props} open={false}>
        Panel contents
      </ContentPickerPanel>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
