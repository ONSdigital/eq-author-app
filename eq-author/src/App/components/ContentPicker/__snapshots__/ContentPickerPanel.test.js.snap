// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPickerPanel should not render when not open 1`] = `""`;

exports[`ContentPickerPanel should render 1`] = `
<ContentPickerPanel__PanelWrapper
  aria-hidden={false}
  aria-labelledby="some-label"
  id="1"
  role="tabpanel"
>
  <ContentPickerPanel__StyledScrollPane>
    Panel contents
  </ContentPickerPanel__StyledScrollPane>
</ContentPickerPanel__PanelWrapper>
`;
